pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -97,7 +98,8 @@ type GraphBuilder struct {
 	// it is protected by monitorLock.
 	running bool
 
-	eventRecorder record.EventRecorder
+	eventRecorder    record.EventRecorder
+	eventBroadcaster record.EventBroadcaster
```

Comment:
not clear on why we grew an event broadcaster

Reply:
removed

Reply:
I reintroduced this change. Initializing eventBroadcaster in both NewComposedGarbageCollector and NewDependencyGraphBuilder results in leaking the goroutine created by NewBroadcaster. Therefore, it is better to initialize it only in GraphBuilder and pass the same instance to GC.

Reply:
Will accept for now, but needs to be fixed for beta.

Reply:
Sure.